scripts/zoho_crm.py

```diff
@@ -0,0 +1,583 @@
+            {"api_name": "Currency_Type", "index": 12},
+            {"api_name": "Deal_Name", "index": 13},
+            {"api_name": "Stage", "index": 14},
+            {"api_name": "Vertical", "index": 15},
+            {"api_name": "Pipeline", "index": 16},
+            {"api_name": "Type", "index": 17},
+            {"api_name": "Primary_Workflow", "index": 18},
+        ],
+        "Contacts": [
+            {"api_name": "Gooey_User_ID", "index": 0},
+            {"api_name": "Gooey_Admin_Link", "index": 1},
+            {"api_name": "Contact_Name", "index": 2},
+            {"api_name": "Last_Name", "index": 3},
+            {"api_name": "Email", "index": 4},
+            {"api_name": "Phone", "index": 5},
+            {"api_name": "Not_Synced", "index": 6},
+            {"api_name": "Contact_Image", "index": 7},
+            {"api_name": "Gooey_Created_Date", "index": 8, "format": "yyyy-MM-dd"},
+            {"api_name": "Gooey_Handle", "index": 9},
+            {"api_name": "Registered_Date", "index": 10, "format": "yyyy-MM-dd"},
+            {"api_name": "Description", "index": 11},
+            {"api_name": "Company", "index": 12},
+            {"api_name": "Personal_Url", "index": 13},
+            {"api_name": "Account_Lookup", "find_by": "Gooey_ID", "index": 14},
+        ],
+        "Accounts": [
+            {"api_name": "Gooey_ID", "index": 0},
+            {"api_name": "Account_Name", "index": 1},
+            {"api_name": "Picture", "index": 2},
+            {"api_name": "Balance", "index": 3},
+            {"api_name": "Is_Paying", "index": 4},
+            {"api_name": "Gooey_Admin_Link", "index": 5},
+            {"api_name": "Created_Date", "index": 6, "format": "yyyy-MM-dd"},
+            {"api_name": "Updated_At", "index": 7, "format": "yyyy-MM-dd"},
+        ],
+    }
+    return field_mappings.get(module, [])
+
+
+def get_zoho_module_name(module: str) -> str:
+    """
+    :param module: Zoho CRM module name
+    :return: Zoho CRM module API name
+    """
+    module_names = {
+        "Contacts": "Contacts",
+        "Accounts": "Accounts",
+        "Deals": "Deals",
+    }
+    return module_names.get(module, "Contacts")
+
+
+def get_unique_field(module: str) -> str:
+    """
+    :param module: Zoho CRM module name
+    :return: Unique field name for the specified module
+    """
+    unique_fields = {
+        "Contacts": "Gooey_User_ID",
+        "Accounts": "Gooey_ID",
+        "Deals": "Invoice_ID",
+    }
+    return unique_fields.get(module, "ID")
+
+
+class ConfigurableFieldMapper:
+    def __init__(self):
+        """
+        :param mapping_config_path: Path to JSON mapping configuration
+        """
+        self.logger = logging.getLogger(self.__class__.__name__)
+        self.mapping_config = self._load_mapping_config()
+
+    def _load_mapping_config(self) -> Dict:
+        """
+        :return: Mapping configuration dictionary
+        """
+        default_config = {
+            "contact_mapping": {
```

Comment:
contact_mapping, transaction_mapping & workspace_mapping can be 3 top level constants instead of a dynamic string inside a dict generated from a class method which is very un-necessary and adds cruft